float64 headPos
bool isLeft
---
